### PR TITLE
Story5

### PR DESCRIPTION
**Pull request by Kel-E-creator**
Hi Ash, Story5 ready for your review please.

### CODE REVIEW

File: app/Http/Controllers/BookController.php [hunk 1]
```diff
@@ -6,21 +6,37 @@
 use App\Models\Claim;
 use Exception;
 use Illuminate\Http\Request;
+use Illuminate\Database\Eloquent\SoftDeletes;
 
 class BookController extends Controller
 {
-
+    use SoftDeletes;
```
Comment (ashleycoles) [inline app/Http/Controllers/BookController.php:13]: You don't need this here - it only needs to go on the model

File: app/Http/Controllers/BookController.php [hunk 2]
```diff
@@ -89,7 +105,51 @@ public function claimBookById(Int $id, Request $request)
...
+
+        $book->claim;
+
+        if(is_null($book->claim)) {
+            return response()->json([
+                'message' => "Book $id is not currently claimed"
+            ], 400);
+        }
+
+        $claim = $this->claim->where('book_id', '=', $id)->firstOrFail();
```
Comment (ashleycoles) [inline app/Http/Controllers/BookController.php:133]: You should probs put this in a try catch, if for some reason the claim doesn't exist, at the moment an exception will crash the app.

Put it in a try catch, and return the exact same response as your 500 response

File: tests/Feature/ReturnTest.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+    //     try {
+            
+    //     } catch (Exception $e) {
+    //         $response->assertServerError()
+    //         ->assertJson([
+    //            'message' => 'Book 1 was returned'
+    //         ]);
+    //     }
+    // }
```
Comment (ashleycoles) [inline tests/Feature/ReturnTest.php:106]: Remove this commented out test

**Review by ashleycoles — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/BookController.php (hunk 1 above)
Comment (W1z4rdK1tty) [reply app/Http/Controllers/BookController.php:13]: Removed.

File: app/Http/Controllers/BookController.php (hunk 2 above)
Comment (W1z4rdK1tty) [reply app/Http/Controllers/BookController.php:133]: We have now put this into a try catch as recommended.

File: tests/Feature/ReturnTest.php (hunk 3 above)
Comment (W1z4rdK1tty) [reply tests/Feature/ReturnTest.php:106]: Have removed commented out code. Thanks

**Review by ashleycoles — APPROVED** [review]

**Pull request merged**